LDMP/main_widget.py

```diff
@@ -121,56 +137,210 @@ def setupDatasets(self):
     def setupAlgorithmsTree(self):
```

Comment:
This method is huge. Please break this up into smaller pieces so that it becomes more readable.